converter/converter.py

```diff
@@ -0,0 +1,202 @@
+from functools import reduce
+from dataclasses import dataclass, field
+from os import path
+
+
+@dataclass(frozen=True)
+class Geometry(ABC):
+    """Abstract geometry dataclass for DetectConfig."""
+
+    pass
+
+
+@dataclass(order=True, frozen=True)
+class Cylinder(Geometry):
+    """Cylinder geometry dataclass used in DetectConfig."""
+
+    sort_index: int = field(init=False)
+
+    id: str
+    radius: int = 10
+    height: int = 10
+
+    template: str = """Geometry Cyl
+    Name ScoringCylinder
+    R 0.0 10.0 {cyl_nr:d}
+    Z 0.0 30.0 {cyl_nz:d}"""
+
+    def __post_init__(self):
+        object.__setattr__(self, 'sort_index', self.id)
+
+    def __str__(self) -> str:
+        return self.template.format(cyl_nr=self.radius, cyl_nz=self.height)
+
+
+@dataclass(order=True, frozen=True)
+class Mesh(Geometry):
+    """Mesh geometry dataclass used in DetectConfig."""
+
+    sort_index: int = field(init=False)
+
+    id: str
+    x: int = 10
+    y: int = 10
+    z: int = 10
+
+    template: str = """Geometry Mesh
+    Name MyMesh_YZ
+    X -5.0  5.0    {mesh_nx:d}
+    Y -5.0  5.0    {mesh_ny:d}
+    Z  0.0  30.0   {mesh_nz:d}"""
+
+    def __post_init__(self):
+        object.__setattr__(self, 'sort_index', self.id)
+
+    def __str__(self) -> str:
+        return self.template.format(mesh_nx=self.x, mesh_ny=self.y, mesh_nz=self.z)
+
+
+@dataclass
+class BeamConfig:
+    """Class mapping of the beam.dat config file."""
+
+    energy: float = 250.
+    nstat: int = 10000
+
+    beam_template: str = """
+RNDSEED      	89736501     ! Random seed
+JPART0       	2            ! Incident particle type
+TMAX0      	{energy:3.6f}   0.0  ! Incident energy; (MeV/nucl)
+NSTAT       {nstat:d}    -1 ! NSTAT, Step of saving
+STRAGG          2            ! Straggling: 0-Off 1-Gauss, 2-Vavilov
+MSCAT           2            ! Mult. scatt 0-Off 1-Gauss, 2-Moliere
+NUCRE           0            ! Nucl.Reac. switcher: 1-ON, 0-OFF
+"""
+
+    def __str__(self) -> str:
+        return self.beam_template.format(energy=self.energy, nstat=self.nstat)
+
+
+@dataclass
+class MatConfig:
+    """Class mapping of the mat.dat config file."""
+
+    materials: list[int] = field(default_factory=lambda: [276])
+
+    material_template: str = """MEDIUM {idx:d}
+ICRU {mat:d}
+END
+"""
+
+    def __str__(self) -> str:
+        material_strings = [self.material_template.format(idx=idx, mat=mat) for idx, mat in enumerate(self.materials)]
+        return "\n".join(material_strings)
+
+
+@dataclass
+class DetectConfig:
+    """Class mapping of the detect.dat config file."""
+
+    geometries: list[Geometry] = field(default_factory=lambda: [Cylinder(id=0), Mesh(id=0)])
+
+    detect_template = """Output
+    Filename mesh.bdo
+    Geo ScoringCylinder
+    Quantity Dose
+    """
+
+    def __str__(self):
+        detect_strings = [str(geo)
+                          for geo in self.geometries] + [self.detect_template]
+        return "\n".join(detect_strings)
+
+
+@ dataclass
```

Comment:
```suggestion
@dataclass
```

Reply:
Resolved in b951def677fac830fd2757dc6fb68fb4a8c5e5e5